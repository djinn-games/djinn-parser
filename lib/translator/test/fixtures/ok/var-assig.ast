{
  "type": "Program",
  "name": {
    "type": "Identifier",
    "name": "var_assignments",
    "line": 1
  },
  "body": {
    "type": "Body",
    "sentences": [
      {
        "type": "ExpressionSentence",
        "expression": {
          "type": "VarDeclaration",
          "dataType": "int",
          "id": {
            "type": "Identifier",
            "name": "x",
            "line": 3
          },
          "init": {
            "type": "OperationExpression",
            "left": {
              "type": "Literal",
              "line": 3,
              "dataType": "int",
              "value": 5
            },
            "right": {
              "type": "Literal",
              "line": 3,
              "dataType": "int",
              "value": 2
            },
            "operator": "add",
            "line": 3
          },
          "line": 3
        },
        "line": 3
      },
      {
        "type": "ExpressionSentence",
        "expression": {
          "type": "VarDeclaration",
          "dataType": "str",
          "id": {
            "type": "Identifier",
            "name": "txt",
            "line": 4
          },
          "line": 4
        },
        "line": 4
      },
      {
        "type": "ExpressionSentence",
        "expression": {
          "type": "CallExpression",
          "callee": {
            "type": "Identifier",
            "name": "log",
            "line": 5
          },
          "args": [
            {
              "type": "AssignmentExpression",
              "operator": "=",
              "left": {
                "type": "Identifier",
                "name": "txt",
                "line": 5
              },
              "right": {
                "type": "Literal",
                "line": 5,
                "dataType": "str",
                "value": "waka"
              },
              "line": 5
            }
          ],
          "line": 5
        },
        "line": 5
      }
    ],
    "line": 2
  },
  "line": 1
}