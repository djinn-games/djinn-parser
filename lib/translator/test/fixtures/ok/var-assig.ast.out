{
  "type": "Program",
  "body": [
    {
      "type": "ExpressionStatement",
      "expression": {
        "type": "AssignmentExpression",
        "operator": "=",
        "left": {
          "type": "MemberExpression",
          "object": {
            "type": "MemberExpression",
            "object": {
              "type": "Identifier",
              "name": "DJINN"
            },
            "property": {
              "type": "Identifier",
              "name": "__scope"
            }
          },
          "property": {
            "type": "Literal",
            "value": "_x",
            "raw": "\"_x\""
          },
          "computed": true
        },
        "right": {
          "type": "BinaryExpression",
          "operator": "+",
          "left": {
            "type": "Literal",
            "value": 5,
            "raw": "5"
          },
          "right": {
            "type": "Literal",
            "value": 2,
            "raw": "2"
          }
        }
      }
    },
    {
      "type": "ExpressionStatement",
      "expression": {
        "type": "AssignmentExpression",
        "operator": "=",
        "left": {
          "type": "MemberExpression",
          "object": {
            "type": "MemberExpression",
            "object": {
              "type": "Identifier",
              "name": "DJINN"
            },
            "property": {
              "type": "Identifier",
              "name": "__scope"
            }
          },
          "property": {
            "type": "Literal",
            "value": "_txt",
            "raw": "\"_txt\""
          },
          "computed": true
        },
        "right": {
          "type": "Literal",
          "value": "",
          "raw": "\"\""
        }
      }
    },
    {
      "type": "ExpressionStatement",
      "expression": {
        "type": "CallExpression",
        "callee": {
          "type": "MemberExpression",
          "object": {
            "type": "Identifier",
            "name": "DJINN"
          },
          "property": {
            "type": "Identifier",
            "name": "log"
          }
        },
        "arguments": [
          {
            "type": "AssignmentExpression",
            "operator": "=",
            "left": {
              "type": "MemberExpression",
              "object": {
                "type": "MemberExpression",
                "object": {
                  "type": "Identifier",
                  "name": "DJINN"
                },
                "property": {
                  "type": "Identifier",
                  "name": "__scope"
                }
              },
              "property": {
                "type": "Literal",
                "value": "_txt",
                "raw": "\"_txt\""
              },
              "computed": true
            },
            "right": {
              "type": "Literal",
              "value": "waka",
              "raw": "\"waka\""
            }
          }
        ]
      }
    }
  ]
}
